conf_scan.l

```diff
@@ -0,0 +1,34 @@
+%{
+
+#include <string.h>
+#include "conf_parse.h"
+
+%}
+
+%option yylineno
+%option noinput nounput
+%option prefix="conf_"
+%option noyywrap
+
+WORD   			[^[:space:][:cntrl:]#]+
+BLANKS			[[:blank:]]+
+%%
+
+{WORD}							{ conf_lval.str_type = strdup(yytext); return T_WORD; }
+
+
+(#.*)+$							/*	ignore comments till the end of a line
+									we need to separately catch a new line in this case! */
+									
+{BLANKS}						return T_BLANK;
+
+\\\r?\n							/* Ignore the newline if it's immediately preceeded by a backslash */
+									
+
+^(\r?\n?[[:blank:]]*(#.*)?)+	return T_NEWLINE;  /*  ignore lines that are empty from the beginning
```

Comment:
I think this is handled if you return T_NEWLINE on comment, because empty_statement in the parser can eat the blank/newline combination?

Reply:
You're right, this rule can be left out.